### PR TITLE
Fix "Created new account" event

### PR DESCRIPTION
**Pull request by NoahCarnahan**
"Created new account" was getting triggered in some instances when it should not have been. This PR moves it to a server side event so that it will only fire when appropriate. Also renames it.

FYI @czue

### CODE REVIEW

File: corehq/apps/analytics/tasks.py [hunk 1]
```diff
@@ -63,3 +64,18 @@ def track_built_app_on_hubspot(webuser):
     if vid:
         # Only track the property if the contact already exists.
         _track_on_hubspot(webuser, {'built_app': True})
+
+@task(queue='background_queue', acks_late=True, ignore_result=True)
```
Comment (dimagimon) [inline corehq/apps/analytics/tasks.py]: E302 expected 2 blank lines, found 1

File: corehq/apps/analytics/tasks.py [hunk 2]
```diff
@@ -63,3 +64,18 @@ def track_built_app_on_hubspot(webuser):
...
+    Record an event in KISSmetrics.
+    :param email: The email address by which to identify the user.
+    :param event: The name of the event.
+    :param properties: A dictionary or properties to set on the user.
+    :return:
+    """
+    api_key = ANALYTICS_IDS.get("KISSMETRICS_KEY", None)
+    if api_key:
+        km = KISSmetrics.Client(key=api_key)
+        res = km.record(email, event, properties if properties else {})
```
Comment (dimagimon) [inline corehq/apps/analytics/tasks.py]: F841 local variable 'res' is assigned to but never used

File: corehq/apps/analytics/tasks.py [hunk 3]
```diff
@@ -63,3 +64,19 @@ def track_built_app_on_hubspot(webuser):
     if vid:
         # Only track the property if the contact already exists.
         _track_on_hubspot(webuser, {'built_app': True})
+
+
+@task(queue='background_queue', acks_late=True, ignore_result=True)
+def track_usage(email, event, properties=None):
```
Comment (czue) [inline corehq/apps/analytics/tasks.py]: non-blocking, but might want to use a function name that implies this is using kissmetrics

**Comment by czue** [comment]
kicked build. :+1: when it passes. feel free to address comment if you want too.

File: corehq/apps/analytics/tasks.py (hunk 3 above)
Comment (NoahCarnahan) [inline corehq/apps/analytics/tasks.py]: Yeah, I thought about that. I decided to name it generically to match up with the corresponding frontend function, [`analytics.wokflow`](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/style/templates/style/includes/analytics_all.html#L69) (although I remembered the name of this function wrong, so I'll change this to `track_workflow()`. The thinking here was that if we ever decided to switch away from KISSmetrics, we wouldn't have to change every function call.

Comment (NoahCarnahan) [inline corehq/apps/analytics/tasks.py]: Renames to `track_workflow()`: https://github.com/dimagi/commcare-hq/pull/6946

**Pull request merged**